raster/src/main/scala/geotrellis/raster/io/geotiff/ArraySegmentBytes.scala

```diff
@@ -51,12 +51,9 @@ object ArraySegmentBytes {
             val result = Array.ofDim[Array[Byte]](offsets.size)
 
             cfor(0)(_ < offsets.size, _ + 1) { i =>
-              byteReader.position(offsets(i).toInt)
-              result(i) = byteReader.getSignedByteArray(byteCounts(i).toInt)
+              result(i) = byteReader.getSignedByteArray(byteCounts(i), offsets(i))
```

Comment:
`ByteReader` is meant to be a replacement for `ByteBuffer`, so I wanted the two to have the same methods (or at least have them be as close as possible). Since none of the methods in `ByteReaderExtensions` are in `ByteBuffer`, they were made into their own separate thing.